### DISCUSSION TITLE
Do pipes break within the command feature?

### DISCUSSION THREAD

**Discussion by Tylermarques** [streamdeck-linux-gui/streamdeck-linux-gui#116] [Ideas / Feature Requests]
I'm trying to run a command that is piped a few times, needed to set my output for `pactl`. See below
```shell
pactl list short sinks | grep iec958 |head -n 1| cut -f1| xargs -I % pactl set-default-sink %
```
I have been having trouble with this for a while, but I finally made it work by changing line 196 in gui.py to `Popen(command, shell=True)`. I have seen some mention of `shell=True` being bad practice but that mostly seems to be due to issues with executing arbitrary code, which is exactly what we want here. 

If anyone has any insight on how I can do this currently, or not require this change please let me know.

**Comment by coolapso** [comment]
Yes .... Put that on a file. 

```
#!/bin/bash 
pactl list short sinks | grep iec958 |head -n 1| cut -f1| xargs -I % pactl set-default-sink %
```

And execute the file instead with `bash /path/to/your/file` 

This is for example how I do it: 

```
#!/bin/bash


speakers=$(pactl list sinks | grep Name | awk '/extra3/ {print $2}')
headphones=$(pactl list sinks | grep Name | awk '/stereo-game/ {print $2}')

if [[ $1 == "speakers" ]]; then
  pactl set-default-sink "$speakers"
fi

if [[ $1 == "headphones" ]]; then
  pactl set-default-sink "$headphones"
fi
```

Then I have 2 states in the button and each one calls the script with the correct argument.

State 1, "command": "bash /home/f0rs3ti/Devel/scripts/global/setAudio.sh headphones",  & Switch to state 2
State 2, "command": "bash /home/f0rs3ti/Devel/scripts/global/setAudio.sh speakers", & Switch to state 1

**Reply by Tylermarques** [reply]
Thanks. I'll make an effort to fork and update some documentation on best practices for shell commands. I was mostly under the impression that anything that worked correctly from my shell should work in the `command` box but that isn't the case.

**Reply by coolapso** [reply]
It will be a very welcome addition!!
